src/deepEqual.js

```diff
@@ -22,9 +22,41 @@
  * deepEqual({test: {abc: 5}}, {test: {abc: 5, def: 4}}) === false
  *
  */
-
 function deepEqual(a, b) {
-  // write code here
+  if (a === b) {
+    return a === b;
+  } else if ((typeof a === typeof b && typeof a === 'object' && a !== null && b !== null)) {
```

Comment:
you can remove else. That will reduce level of nesting in this function.

Reply:
done